interview-questions.md

```diff
@@ -6,63 +6,108 @@ Answer the following questions. First, without external resources. Challenge you
 
 
 2. What is the difference between map and filter?
 
   Your answer:
+  The difference is that .map() looks at each element in an array while .filter() has a true/false code that your array has to go through.
+
+  .map() is an array method, returns to us a completely new array.
+
+  .filter() is also an array method, like map, but it has a built in if/else statement so it looks for true/false values. It returns back what we "filtered" through it. 
 
   Researched answer:
 
+  They both return a new array. map returns a new array of elements where you have applied some function on the element so that it changes the original element (typically). filter returns a new array of the elements of the original array (with no change to the original values). filter will only return elements where the function you specify returns a value of true for each element passed to the function.
+
+  So map returns the same number of elements as the original, but the element value will be transformed in some way and filter will return the same or less number of elements than the original but not change the original elements’ values.
 
 
 3. What is the difference between map/filter and a for loop?
 
   Your answer:
+  A for loop iterates from a defined start and end point with a defined number of times. Map/filter only go through the array once.
+
 
   Researched answer:
 
+  The big difference is that the conditions for iteration are handled for us. In a for loop, we’re guaranteed to get every element of the array in the right order. We don’t have to manage our own index variable or condition for that variable.
 
 
 4. What is the difference between console.log() and return?
 
   Your answer:
 
+  console.log is used for developers, we shouldn't count on it when defining a function. It's helpful when testing out a piece of code. A return would live inside a function and can be used in reference inside the function.
+
   Researched answer:
 
+  Console.log is a function that logs the arguments you pass to the web console. return is a statement which specifies the value returned from a function.
 
 
 5. In regards to functions, what is an argument?
 
   Your answer:
 
+  Parameters. They are used in a function as placeholders for information.
+
   Researched answer:
 
+  The parameters, in a function call, are the function's arguments. JavaScript arguments are passed by value. The function only gets to know the values, not the argument's locations. If a function changes an argument's value, it does not change the parameter's original value.
 
 
 6. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
   Your answer:
 
+  Pair programming allows for partners to share and equally participate when coding. They take turns being the driver, where a partner is working on the live code. The driver then pushes the code up for the other partner to have access to it. It is about working together on a single project.
```

Comment:
Just about right, I think I'd mention the navigator role here too, as the question asks specifically about the roles of each person.